frame/oracle/src/lib.rs

```diff
@@ -692,11 +696,25 @@ pub mod pallet {
 				let min_accuracy = asset_info.threshold;
 				if accuracy < min_accuracy {
 					let slash_amount = asset_info.slash;
-					let try_slash = T::Currency::can_slash(&answer.who, slash_amount);
-					if !try_slash {
-						log::warn!("Failed to slash {:?}", answer.who);
+					let new_amount_staked = Self::oracle_stake(answer.who.clone())
+						.unwrap_or_else(|| 0_u32.into())
+						.saturating_sub(slash_amount);
+					OracleStake::<T>::insert(&answer.who, new_amount_staked);
+					let result = T::Currency::repatriate_reserved(
+						&answer.who,
+						&T::TreasuryAccount::get(),
+						slash_amount,
+						BalanceStatus::Free,
+					);
+					match result {
+						Ok(remaning_val) =>
+							if remaning_val > BalanceOf::<T>::zero() {
+								log::warn!("Only slashed {:?}", slash_amount - remaning_val);
+							},
+						Err(e) => {
+							log::warn!("Failed to slash {:?} due to {:?}", answer.who, e);
+						},
 					}
```

Comment:
Fine to log this but I believe it's impossible as you are checking that the staked amount >= asset slash right?

Reply:
Yes I think this is correct, that change is done in different PR.

Reply:
But it is fine just for sake of completion.